fix(test): fix unit tests

cwillisf opened the pull request:
### Resolves

Resolves #2528
Resolves #2527
Might be related to #1154

### Proposed Changes

If any of the calls to the Selenium web driver's `browser` object fail with a promise rejection, the test runner now reports the error and exits with code 1. To make this easier to read, I've converted the body of the test into an `async` function which uses `await` when making these calls.

UPDATE: I've added changes which actually make the tests pass, including:
- Update the `chromedriver` dependency
- Disable the Chrome sandbox on Linux CI
- Enable Chrome's headless mode on CI regardless of platform
- Quit the browser session after tests

### Reason for Changes

The previous formulation of the test runner was not catching promise rejections coming from the webdriver / browser. Node would print warnings to the console, but the test runner would exit with code 0 and CI would treat this as a passing build. Now the rejection is caught and reported, and the test runner exits with code 1.

Existing functionality to look for failed test results has not changed.

UPDATE:
- Updating the `chromedriver` dependency was needed to get the tests running at all. In fact, this was enough to get them working for me locally.
- I disabled the Chrome sandbox on Linux CI to get the tests working on Travis. See [Travis docs](https://docs.travis-ci.com/user/chrome#sandboxing)
- Chrome's headless mode is kind of a nice-to-have in general, but may be necessary on Travis since XVFB doesn't seem to be working.
- If we don't quit the browser session after tests, Chrome processes are left running each time. This isn't a big deal when testing on CI since the VM is short-lived anyway, but it's annoying when running tests locally.

### Test Coverage

The first commit in this PR made it so that test failures would be noticed on CI, so the fact that it failed with ❌ is a success. From there on, getting the existing tests to pass was the goal.

cwillisf committed:
fix(test): treat failure to launch tests as a test failure

The previous formulation of the test runner was not catching promise
rejections coming from the webdriver / browser. Node would print
warnings to the console, but the test runner would exit with code 0 and
CI would treat this as a passing build. Now the rejection is caught and
reported, and the test runner exits with code 1.

Existing functionality to look for failed test results has not changed.

rschamp reviewed (commented):
So, we should approve this because it's failing the tests? 🙃  L❌TM
Thanks for catching this, it does seem related to the problem in #1154.  If we merge this, then we won't be able to release scratch-blocks until we fix the tests... Maybe we should fix them in a separate branch based on this one?

cwillisf commented:
Yeah, I was thinking about this last night and came to the same conclusion: as bad as it is to ignore passing tests, it's probably worse to stop publishing altogether _in this particular case_ since the codebase is stable and it's just localization that's changing regularly. I'll try to figure out what's going wrong -- maybe just a Selenium update.

cwillisf committed:
fix(test): bump chromedriver to get unit tests working again

cwillisf committed:
fix(test): quit Selenium session after tests

cwillisf committed:
fix(test): fix tests on Travis by disabling Chrome sandbox on Linux CI

Travis documentation suggests that disabling the sandbox might be
necessary to get Chrome to run.

rschamp reviewed (approved):
Nice work!
